### PR TITLE
Fix cisco standard header comment

### PR DESCRIPTION
**Pull request by zombah**
Cisco standard header comment must follow standard remark syntax

### CODE REVIEW

File: lib/cisco.py [hunk 1]
```diff
@@ -999,7 +999,11 @@ def __str__(self):
         # add a header comment if one exists
         for comment in header.comment:
           for line in comment.split('\n'):
-            target.append(' remark %s' % line)
+            if filter_type == 'standard':
+              target.append(' access-list %s remark %s' % (filter_name,
+                                                           line))
```
Comment (ankenyr) [inline lib/cisco.py]: Please put this on one line since it doesn't go over the line limit of 80 characters

**Comment by zombah** [comment]
Here is how i noticed this problem,
Example policy:
```
header {
  comment:: "Test policy for cisco standard filter"
  target:: cisco 66 standard
}
term allow-ntp {
  comment:: "Allow example ntp servers"
  address:: NTP_SERVERS
  action:: accept
}
term allow-tacacs {
  address:: TACACS_SERVERS
  action:: accept
}
```

Current resulting filter:  
```
! $Id:$
! $Date:$
! $Revision:$
no access-list 66
access-list 66 remark $Id:$
 remark Test policy for cisco standard filter
access-list 66 remark allow-ntp
access-list 66 remark Allow example ntp servers
access-list 66 permit 10.0.0.1
access-list 66 permit 10.0.0.2
access-list 66 remark allow-tacacs
access-list 66 permit 10.1.0.1
access-list 66 permit 10.1.0.2

exit
```

Here is cisco output while such filter applied:  
```
test.router#conf t
Enter configuration commands, one per line.  End with CNTL/Z.
test.router(config)#! $Id:$
test.router(config)#! $Date:$
test.router(config)#! $Revision:$
test.router(config)#no access-list 66
test.router(config)#access-list 66 remark $Id:$
test.router(config)# remark Test policy for cisco standard filter
                          ^
% Invalid input detected at '^' marker.

test.router(config)#access-list 66 remark allow-ntp
test.router(config)#access-list 66 remark Allow example ntp servers
test.router(config)#access-list 66 permit 10.0.0.1
test.router(config)#access-list 66 permit 10.0.0.2
test.router(config)#access-list 66 remark allow-tacacs
test.router(config)#access-list 66 permit 10.1.0.1
test.router(config)#access-list 66 permit 10.1.0.2
test.router(config)#
test.router(config)#exit
```

this probably not a big deal with manual filter distribution, but for automated tasks cli stdout error is not very good.

**Comment by zombah** [comment]
Documentation about remark in standard access-list: https://www.cisco.com/c/en/us/td/docs/ios/12_2/ipaddr/command/reference/fipras_r/1rfip1.html#wp1029863

**Comment by ankenyr** [comment]
I believe v12.2 has been end of life for years now. The documentation for >v15 provides the format above.

I don't think we want to accept this PR since it could break anyone on current versions.

**Comment by morrowc** [comment]
zombah - note that 12.2 is .. long eol/eos from cisco. On a current 15.x device:
 Version 15.1(2)SG7,

rr11.atl11(config)#no access-list 66
rr11.atl11(config)#access-list 66 remark $Id:$
rr11.atl11(config)# remark Test policy for cisco standard filter
                      ^
% Invalid input detected at '^' marker.

rtr(config)#access-list 66 remark allow-ntp
rtr(config)#access-list 66 remark Allow example ntp servers
rtr(config)#access-list 66 permit 10.0.0.1
rtr(config)#access-list 66 permit 10.0.0.2
rtr(config)#access-list 66 remark allow-tacacs
rtr(config)#access-list 66 permit 10.1.0.1
rtr(config)#access-list 66 permit 10.1.0.2
rtr(config)#no access-list 66
rtr(config)#exit

**Comment by morrowc** [comment]
it's possible that this is a problem with 'standard' vs 'extended' access-lists though.

rtr(config)#no access-list 166
rtr(config)#access-list 166 remark $Id: $
rtr(config)#access-list 166 permit ip any any
rtr(config)#access-list 166 remark "Some remark here"
rtr(config)#access-list 166 deny ip any any
rtr(config)#exit
rtr#

yes, best guess is: "Standard access list bug"

maybe open a 'your config should work consistently' bug to cisco? :)

**Comment by zombah** [comment]
@morrowc 
166 is extended access-list not standard, you can check that in ios cli with access-list ?:
```
  <1-99>            IP standard access list
  <100-199>         IP extended access list
```
I see no difference in standard behaiviour even on latest 15.x versions here.

**Comment by zombah** [comment]
@ankenyr Here is doc for current versions, i see no presence of syntax without "access-list [1-99]" for standard acl
https://www.cisco.com/c/en/us/support/docs/security/ios-firewall/23602-confaccesslists.html?dtid=osscdc000283#standacl

**Comment by morrowc** [comment]
@zombah so, yes I get that 166 is 'extended' not 'standard', my point was that for an access list as you demonstrated (standard) the remark ordering seems to matter. For an extended list (166) the ordering doesn't seem to matter (for the version of IOS I was testing on).

Were you seeing something different? I've not seen a 12.2 box in ~5+yrs so I can't test on that version.

**Comment by zombah** [comment]
@morrowc Standard acl only works with access-list prefix and extended have syntax versions without prefix thats why i added condition for standard variant only not for others. It is same for 15.x versions  or ios-xe. I see no cisco bug here, they just keep standard acl backward compatibility.

**Comment by morrowc** [comment]
okey dokey

**Comment by zombah** [comment]
@morrowc @ankenyr So you see now that my fix correct or i need more explanation?

**Comment by ankenyr** [comment]
Sure this seems fine to me now, please resolve the conflicts and I can pull it in.

**Comment by zombah** [comment]
@ankenyr As i understand conflict is with pendending pr #152? Once it will be merged i'll rebase my commit or vice versa.

**Comment by ankenyr** [comment]
Ahh ok so you based yours off of his. If you wish to wait that is fine.

**Comment by zombah** [comment]
No i based versus master and i don't see conflicts while diffing my branch vs upstream/master. So that was my idea that github also checking PR's

**Comment by zombah** [comment]
Ops no i'm wrong my upstream is outdated, i'll fix conflict

**Comment by ankenyr** [comment]
You shouldn't need to wait. I do not know how long orbit will take to
respond.

**Comment by zombah** [comment]
@ankenyr I resolved conflicts

**Comment by ankenyr** [comment]
Please feel free to redo this change but lets modify it so that it only gets outputted on the cisco platform and not any of the ones that are inherited.

**Pull request merged**